crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -74,29 +77,39 @@ impl TradeVerifier {
     pub fn new(
         simulator: Arc<dyn CodeSimulating>,
         code_fetcher: Arc<dyn CodeFetching>,
+        trusted_tokens: Arc<AutoUpdatingTokenList>,
         settlement: H160,
         native_token: H160,
     ) -> Self {
         Self {
             simulator,
             code_fetcher,
+            trusted_tokens,
             settlement,
             native_token,
         }
     }
-}
 
-#[async_trait::async_trait]
-impl TradeVerifying for TradeVerifier {
-    async fn verify(
+    async fn verify_inner(
         &self,
         query: &PriceQuery,
         verification: &Verification,
-        trade: Trade,
+        trade: &Trade,
+        internalize: bool,
     ) -> Result<VerifiedEstimate> {
         let solver = dummy_contract!(Solver, trade.solver);
 
-        let settlement = encode_settlement(query, verification, &trade, self.native_token);
+        let interactions = match internalize {
+            true => &[],
```

Comment:
The solver may not want all interactions to be internalised (e.g. PMMs) and thus we might internalize things here which later won't be internalizable in the solution we receive during solving.